arangod/Aql/QueryOptions.cpp

```diff
@@ -103,6 +104,12 @@ void QueryOptions::fromVelocyPack(VPackSlice const& slice) {
     maxWarningCount = value.getNumber<size_t>();
   }
 
+  value = slice.get("timeout");
+  if (value.isNumber()) {
+    timeout = value.getNumber<size_t>();
```

Comment:
let's make this a double, so you can specify fractional seconds (even though they will not be honored completely accurately).